regions/shapes/moc/core.py

```diff
@@ -0,0 +1,941 @@
+            # overlap area of the total MOC
+            fraction_sum += fraction
+
+        # Clip its values to the interval (0, 1)
+        fraction_clipped = np.clip(a=fraction_sum, a_min=0, a_max=1)
+        return RegionMask(fraction_clipped, bbox=bbox)
+
+    def contains(self, pixcoord):
+        """
+        Checks whether a position or positions fall inside the region.
+
+        Parameters
+        ----------
+        pixcoord : `~regions.PixCoord`
+            The position or positions to check.
+        """
+        pixcoord = PixCoord._validate(pixcoord, 'pixcoord')
+        x = np.atleast_1d(np.asarray(pixcoord.x, dtype=float))
+        y = np.atleast_1d(np.asarray(pixcoord.y, dtype=float))
+
+        shape = x.shape
+        vx, vy = self.vertices_culled
+        mask = points_in_polygons(x.flatten(), y.flatten(), vx, vy).astype(bool)
+        in_poly = mask.reshape(shape)
+        if self.meta.get('include', True):
+            return in_poly
+        else:
+            return np.logical_not(in_poly)
+
+class MOCSkyRegion(SkyRegion):
+    """
+    A MOC (Multi-Order Coverage map)
+
+    MOC stands for Multi-Order Coverage. It is a spatial and hierarchical description of a region on a sphere.
+    MOC is an IVOA standard which was first introduced in the following
+    `paper <http://www.ivoa.net/documents/MOCSkyRegion.>`__ .
+    MOC are based on the HEALPix sky tessellation using the NESTED numbering scheme. A MOC is a set of
+    HEALPix cells at different orders with a maximum resolution corresponding to the order 29 i.e. a cell
+    resolution of ~393.2uas.
+
+    * MOCs are usually stored as FITS file containing a list of UNIQ numbers describing the HEALPix cells at \
+      different orders. This class aims at creating MOC maps from FITS/json formatted files, FITS images associated with a mask numpy array, \
+      an `astropy.coordinates.SkyCoord` object and lon, lat expressed as `astropy.units.Quantity` objects.
+
+    * Basic operations on MOCs are available such as the intersection, union, difference and complement of a MOC.
+
+    * A :meth:`~regions.MOCSkyRegion.contains` method filters positions expressed as \
+      `~astropy.units.Quantity` to only keep those lying inside/outside the MOC.
+
+    * A MOC can be serialized to FITS (i.e. a list of UNIQ numbers stored in a \
+      binary HDU table) and JSON formats. An optional parameter allows the user to write it to a file.
+
+    Parameters
+    ----------
+    interval_set : `~regions.IntervalSet` object, optional
+        A N rows by 2 columns `~numpy.ndarray` storing the set of intervals
+        representing the data structure of the MOCSkyRegion.
+    meta : `~regions.RegionMeta` object, optional
+        A dictionary which stores the meta attributes of this region.
+    visual : `~regions.RegionVisual` object, optional
+        A dictionary which stores the visual meta attributes of this region.
+
+    Examples
+    --------
+    .. plot::
```

Comment:
Personally I think a new high-level page `docs/moc.rst` that explains a bit what MOC is and gives examples like this one would be nicer than having long examples / plots in the API docs.
Note that it's easy to create links back and forth, for sure the class docstrings should contain a link to the high-level docs page.